### PR TITLE
ncnn master

### PR DESCRIPTION
**Pull request by fantes**
this PR makes dd use tencent/master as source of ncnn, instead of our old fork

### CODE REVIEW

File: src/backends/ncnn/ncnnlib.cc [hunk 1]
```diff
@@ -85,7 +85,10 @@ namespace dd
   NCNNLib<TInputConnectorStrategy, TOutputConnectorStrategy,
           TMLModel>::~NCNNLib()
   {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdelete-non-virtual-dtor"
```
Comment (BynaryCobweb) [inline src/backends/ncnn/ncnnlib.cc:89]: Do you know where this warning comes from?

**Review by BynaryCobweb — APPROVED** [review]

File: src/backends/ncnn/ncnnlib.cc (hunk 1 above)
Comment (fantes) [reply src/backends/ncnn/ncnnlib.cc:89]: somthing like that :```

**Comment by beniz** [comment]
Great! I believe we may want to add a test on a TS model at this stage, to keep track of how our modified LSTM holds within NCNN.

**Comment by beniz** [comment]
@fantes This PR does not work anymore with latest ncnn, see error below:
```
In file included from /home/beniz/projects/deepdetect/dev/deepdetect/build_ncnn/ncnn/src/ncnn/src/c_api.cpp:26:0:
/home/beniz/projects/deepdetect/dev/deepdetect/build_ncnn/ncnn/src/ncnn/src/net.h: In member function ‘void ncnn::Net::set_input_h(int) const’:
/home/beniz/projects/deepdetect/dev/deepdetect/build_ncnn/ncnn/src/ncnn/src/net.h:79:17: error: invalid types ‘<unresolved overloaded function type>[int]’ for array subscript
         layers[0]->set_h(h);
```

In all cases, after fixing the patch on our side, I suggest to pin the NCNN version to the latest release with `GIT_TAG` in the cmakelist file, e.g. https://github.com/Tencent/ncnn/releases

**Comment by fantes** [comment]
> Great! I believe we may want to add a test on a TS model at this stage, to keep track of how our modified LSTM holds within NCNN.

done !

**Comment by fantes** [comment]
> @fantes This PR does not work anymore with latest ncnn, see error below:
> 
> ```
> In file included from /home/beniz/projects/deepdetect/dev/deepdetect/build_ncnn/ncnn/src/ncnn/src/c_api.cpp:26:0:
> /home/beniz/projects/deepdetect/dev/deepdetect/build_ncnn/ncnn/src/ncnn/src/net.h: In member function ‘void ncnn::Net::set_input_h(int) const’:
> /home/beniz/projects/deepdetect/dev/deepdetect/build_ncnn/ncnn/src/ncnn/src/net.h:79:17: error: invalid types ‘<unresolved overloaded function type>[int]’ for array subscript
>          layers[0]->set_h(h);
> ```
> 
> In all cases, after fixing the patch on our side, I suggest to pin the NCNN version to the latest release with `GIT_TAG` in the cmakelist file, e.g. https://github.com/Tencent/ncnn/releases

done !

**Review by beniz — APPROVED** [review]

**Pull request merged**